libraries/AP_Scripting/drivers/Hobbywing_DataLink.lua

```diff
@@ -0,0 +1,231 @@
+    { 97, 	80}, 	{ 95, 	81}, 	{ 93, 	82}, 	{ 91, 	83}, 	{ 90, 	84}, 	{ 88, 	85}, 	{ 85, 	86}, 	{ 84, 	87}, 	{ 82, 	88}, 	{ 81, 	89},
+    { 79, 	90}, 	{ 77, 	91}, 	{ 76, 	92}, 	{ 74, 	93}, 	{ 73, 	94}, 	{ 72, 	95}, 	{ 69, 	96}, 	{ 68, 	97}, 	{ 66, 	98}, 	{ 65, 	99},
+    { 64, 	100}, 	{ 62, 	101}, 	{ 62, 	102}, 	{ 61, 	103}, 	{ 59, 	104}, 	{ 58, 	105}, 	{ 56, 	106}, 	{ 54, 	107}, 	{ 54, 	108}, 	{ 53, 	109},
+    { 51, 	110}, 	{ 51, 	111}, 	{ 50, 	112}, 	{ 48, 	113}, 	{ 48, 	114}, 	{ 46, 	115}, 	{ 46, 	116}, 	{ 44, 	117}, 	{ 43, 	118}, 	{ 43, 	119},
+    { 41, 	120}, 	{ 41, 	121}, 	{ 39, 	122}, 	{ 39, 	123}, 	{ 39, 	124}, 	{ 37, 	125}, 	{ 37, 	126}, 	{ 35, 	127}, 	{ 35, 	128}, 	{ 33, 	129},
+}
+
+local function temperature_decode(temp_raw)
+   if temp_raw == 0 then
+      return 0
+   end
+   for i = 1, #temp_table do
+      if temp_table[i][1] <= temp_raw then
+         return temp_table[i][2]
+      end
+   end
+   return 130
+end
+
+local function decode_current(curr)
+   if curr == 0xffff then
+      return 0.0
+   end
+   return curr / 64.0
+end
+
+local telem_data = ESCTelemetryData()
+
+--[[
+   check for input and parse data
+--]]
+local function check_input()
+   local n_bytes = uart:available():toint()
+   if n_bytes < 160 then
+      return
+   end
+   --gcs:send_text(0,string.format("n_bytes=%u", n_bytes))
+   if n_bytes > 160 then
+      discard_pending()
+      return
+   end
+
+   local s = read_bytes(n_bytes)
+   local head, frame_len, ver, cmd, _ = string.unpack(">BBBBH", string.sub(s,1,6))
+   if head ~= 0x9B or frame_len ~= 158 or ver ~= 1 or cmd ~= 2 then
+      gcs:send_text(MAV_SEVERITY.INFO, string.format("bad frame %x %x %x %x", head, frame_len, ver, cmd))
+      return
+   end
+   local crc1 = string.unpack("<H", string.sub(s,159,160))
+   local crc2 = crc_xmodem(string.sub(s,1,158))
+   if crc1 ~= crc2 then
+      -- gcs:send_text(MAV_SEVERITY.INFO, string.format("bad crc %x %x", crc1, crc2))
+      return
+   end
+   for i = 0, 7 do
+      local e = string.sub(s,7+i*19,25+i*19)
+      local _, pnum, in_thr, out_thr, eRPM, volt, curr, pcurr, mos_temp, cap_temp, status = string.unpack(">BHHHHHHHBBH", e)
+      local RPM = math.floor(eRPM*10.0/ESC_HW_POLES:get())
+      if volt > 0 or curr > 0 or RPM > 0 or pnum > 1 then
+         -- we have valid ESC data
+         local ofs = ESC_HW_OFS:get()
+         curr = decode_current(curr)
+         pcurr = decode_current(pcurr)
+         volt = volt * 0.1
+         in_thr = in_thr / 32768.0
+         out_thr = out_thr / 32768.0
+         mos_temp = temperature_decode(mos_temp)
+         cap_temp = temperature_decode(cap_temp)
+         telem_data:voltage(volt)
+         telem_data:current(curr)
+         telem_data:temperature_cdeg(math.floor(mos_temp*100))
+         esc_telem:update_rpm(ofs+i, math.floor(eRPM*10.0/ESC_HW_POLES:get()), 0)
+         -- 0x0D is temperature + voltage + current
```

Comment:
With custom bindings we could make setting of the mask automatic. A valid value would add the bit to the mask and setting to nil would clear it.

Probably not worth the effort tho.